lib/helpers/getDomNodeAttributes.js

```diff
@@ -0,0 +1,14 @@
+export function getDomNodeAttributes(element,filter) {
+  let i,
+    attributeNodes = element.attributes || {},
+    length = attributeNodes.length,
+    attrs = {};
+
+  for ( i = 0; i < length; i++ ){
+    if(filter == '' || filter.test(attributeNodes[i].name) == true) {
+     attrs[attributeNodes[i].name] = attributeNodes[i].value;
```

Comment:
can consider making a variable for `attributeNodes[i]` since it's used three times and to avoid traversing attributeNodes repeatedly